util/template/expression_template.go

```diff
@@ -35,3 +42,17 @@ func envMap(replaceMap map[string]string) map[string]interface{} {
 	}
 	return envMap
 }
+
+// hasRetries checks if the variable `retries` exists in the expression template
+func hasRetries(expression string) bool {
+	tokens, err := lexer.Lex(file.NewSource(expression))
+	if err != nil {
+		return false
+	}
+	for _, token := range tokens {
+		if token.Kind == lexer.Identifier && token.Value == "retries" {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Can't really use a simple string matching method like `strings.Contains` since we don't want this logic to be applied on user-defined variables( like `retriesCustom`), so I have to call the `lexer` package, similar to what's being done in `expr`: https://github.com/antonmedv/expr/blob/d389a61c1333215fad33523ca988970f6e77a2b6/parser/parser.go#L89-L91

Reply:
Great find